marketplace/deployer_util/expand_config.py

```diff
@@ -84,18 +91,28 @@ def expand(values_dict, schema, app_uid=''):
         v = True
       elif prop.default is not None:
         v = prop.default
-    else:  # if v is None
+
+    # The value is not empty, so it can be expanded to the special properties.
+    # For example, property IMAGE can be expanded from the raw string image,
+    # to the generatedProperties properties.
+    if v is not None:
       if prop.image:
         if not isinstance(v, str):
           raise InvalidProperty(
               'Invalid value for IMAGE property {}: {}'.format(k, v))
         generate_properties_for_image(prop, v, generated)
-      if prop.string:
+      elif prop.string:
         if not isinstance(v, str):
           raise InvalidProperty(
               'Invalid value for STRING property {}: {}'.format(k, v))
         generate_properties_for_string(prop, v, generated)
+      elif prop.tls_certificate:
+        if not isinstance(v, str):
+          raise InvalidProperty(
+              'Invalid value for TLS_CERTIFICATE property {}: {}'.format(k, v))
+        generate_properties_for_tls_certificate(prop, v, generated)
 
+    # At this point, the property is populated and expanded, so overwrite the returned value.
```

Comment:
This is not overriding--this is recording the value of the property. I think we need to clarify what `result` and `generated` dictionaries capture. Can you update L71-73 and L122 with the following:

```python
### L71-73 ###

# Captures the final property name-value mappings.
# This has both properties directly specified under schema's `properties` and
# generated properties. See below for details about generated properties.
result = {}
# Captures only the generated properties. These are not directly specified in the schema
# under `properties`. Rather, their name are specified in special `generatedProperties` fields
# under individual property `x-google-marketplace`.
# Note that properties with generated values are NOT generated properties. 
generated = {}

# Copy explicitly specified values and generate values into result.

### L122 ###
# Copy generated properties into result, validating that there are no collisions.
```

Reply:
Done.